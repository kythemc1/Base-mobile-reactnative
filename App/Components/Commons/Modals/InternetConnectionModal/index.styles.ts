import {StyleSheet} from 'react-native';
import {colors} from 'Configs/UI/Colors';
import sizes from 'Configs/UI/Sizes';

export const styles = StyleSheet.create({
    mContainer: {
        backgroundColor: colors.color_white,
        width: sizes._screenWidth * 0.9,
        borderRadius: sizes._10sdp,
        padding: sizes._16sdp,
        height: sizes._screenWidth * 0.9,
        alignItems: 'center',
        justifyContent: 'space-evenly'
    },
    mTitle: {
        fontSize: sizes._20sdp,
        fontWeight: '600',
        textAlign: 'center',
        lineHeight: sizes._28sdp,
        fontFamily: 'LexendDeca-Bold',
        color: colors.color_primary1
    },
    mText: {
        textAlign: 'center',
        color: colors.color_grey2,
        fontSize: sizes._14sdp,
        fontWeight: '400',
        lineHeight: sizes._20sdp,
        fontFamily: 'LexendDeca-Light'
    },
    img: {
        width: sizes._160sdp,
        height: sizes._160sdp
    }
});
