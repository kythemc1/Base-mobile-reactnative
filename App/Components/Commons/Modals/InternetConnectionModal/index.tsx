import Modal from 'react-native-modal';
import React from 'react';
import {Image, Text, View} from 'react-native';
import {styles} from 'Components/Commons/Modals/InternetConnectionModal/index.styles';
import {useTranslation} from 'react-i18next';
import {Button} from 'Components/Commons/Button';

interface Props {
    isConnected: boolean | undefined;
    title: string,
    subtitle: string,
    img?: any;
    onPress?: () => void;
}

export const InternetConnectionModal = (props: Props) => {
    const {t} = useTranslation();
    return (
        <Modal isVisible={props.isConnected}
            statusBarTranslucent={true}
            backdropOpacity={0.4}>
            <View style={styles.mContainer}>
                <Image source={props.img} style={styles.img} resizeMode={'contain'}/>
                <Text style={styles.mTitle}>{props.title}</Text>
                <Text style={styles.mText}>
                    {props.subtitle}
                </Text>
                <View style={{marginHorizontal: 16}}>
                    <Button text={t('internet.title_button')} typeButton={'secondary'} sizeButton={'custom'}
                        onPressButton={props.onPress}/>
                </View>

            </View>
        </Modal>
    );
};
