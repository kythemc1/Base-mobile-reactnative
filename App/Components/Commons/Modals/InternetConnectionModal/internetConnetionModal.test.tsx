import renderer from 'react-test-renderer';
import { InternetConnectionModal } from 'Components/Commons/Modals/InternetConnectionModal/index';
jest.useRealTimers();
test('Test snap modal InternetConnetionModal', () => {
    const screen = renderer.create(<InternetConnectionModal isConnected={true} />).toJSON();
    expect(screen).toMatchSnapshot();
});
