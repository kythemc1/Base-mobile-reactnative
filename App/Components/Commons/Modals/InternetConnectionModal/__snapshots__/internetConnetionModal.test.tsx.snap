// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snap modal InternetConnetionModal 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={true}
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 10,
          "height": 150,
          "justifyContent": "space-evenly",
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#143980",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        internet.title_modal
      </Text>
      <Text
        style={
          {
            "color": "#0E1E38",
            "fontSize": 16,
            "fontWeight": "300",
            "textAlign": "center",
          }
        }
      >
        internet.message_modal
      </Text>
    </View>
  </View>
</Modal>
`;
